cmd/postgres_exporter/postgres_exporter.go

```diff
@@ -517,7 +517,7 @@ func makeQueryOverrideMap(pgVersion semver.Version, queryOverrides map[string][]
 	return resultMap
 }
 
-func parseUserQueries(content []byte) (map[string]intermediateMetricMap, map[string]string, error) {
+func parseUserQueries(content []byte, server *Server) (map[string]intermediateMetricMap, map[string]string, error) {
```

Comment:
`parseUserQueries` should compare the current server version with the version of the custom metric configuration.

It's strange that you make logical judgments in this loop.